src/core/MOM_open_boundary.F90

```diff
@@ -253,6 +265,20 @@ module MOM_open_boundary
                    tracer_y_reservoirs_used => NULL() !< Dimensioned by the number of tracers, set globally,
                                                       !! true for those with y reservoirs (needed for restarts).
   integer                       :: ntr = 0            !< number of tracers
+  integer :: n_tide_constituents = 0                  !< Number of tidal constituents to add to the boundary.
+  logical :: add_tide_constituents = .false.          !< If true, add tidal constituents to the boundary elevation
+                                                      !! and velocity. Will be set to true if n_tide_constituents > 0.
+  character(len=2), allocatable, dimension(:) :: tide_names  !< Names of tidal constituents to add to the boundary data.
+  real, allocatable, dimension(:) :: tide_frequencies        !< Angular frequencies of chosen tidal constituents.
```

Comment:
The units of all real elements in public types are supposed to be documented in the comments describing them. In the cast of tide_frequencies, this might be `[s-1]` or `[T-1 ~> s-1]`, depending on whether these units have been dimensionally rescaled.  Similarly the units tide_eq_phases, tide_fn, tide_un, time_ref, and astro_shpn need to be documented here. If they are nondimensional, the convention is to describe their units as `[nondim]`.